packages/@contentful--app-scripts/lib/upload/organization-api.js

```diff
@@ -0,0 +1,72 @@
+const inquirer = require('inquirer');
+const chalk = require('chalk');
+const ora = require('ora');
+
+async function fetchOrganizations(client) {
+  try {
+    const orgs = await client.getOrganizations();
+
+    return orgs.items.map((org) => ({
+      name: org.name,
+      value: org.sys.id,
+    }));
+  } catch (err) {
+    console.log(`
+${chalk.red(
+  'Error:'
+)} Could not fetch your organizations. Make sure you provided a valid access token.
+
+${err.message}
+    `);
+
+    throw err;
+  }
+}
+
+async function selectOrganization(client) {
+  const orgSpinner = ora('Fetching your organizations').start();
+  let organizations = null;
+  try {
+    organizations = await fetchOrganizations(client);
+  } finally {
+    orgSpinner.stop();
+  }
+
+  if (!organizations) {
+    return null;
+  }
+  const { organizationId } = await inquirer.prompt([
+    {
+      name: 'organizationId',
+      message: 'Select an organization for your app:',
+      type: 'list',
+      choices: organizations,
+    },
+  ]);
+  return organizations.find((org) => org.value === organizationId);
+}
```

Comment:
Selecting from a list is something we do quite often. I would extract a utilty for it and reuse it for definitions and organizaions

Reply:
nice idea I abstracted it in utils 👍